# 👥 Population and Sample

In Statistics, we often want to understand a large group of people, objects, or observations.

Studying the entire group may not always be practical.

Therefore, we often select a smaller group from the larger group and use it to learn about the whole.

This notebook introduces:

- 👥 Population
- 🎯 Sample
- 📐 Parameter
- 📊 Statistic
- 🔍 Population vs Sample
- 🎲 Why Sampling is necessary
- 🤖 How Population and Sample are used in AI/ML

## 🎯 Learning Objectives

By the end of this notebook, you will understand:

- What a Population is
- What a Sample is
- The difference between Population and Sample
- What Parameters and Statistics mean
- Why we use samples instead of studying entire populations
- What makes a sample useful
- How sampling is connected to Data Science and Machine Learning

## 👥 What is a Population?

In Statistics, a **Population** refers to the complete group of individuals, objects, or observations that we are interested in studying.

The population does not necessarily mean "people".

It can be:

- 👨‍🎓 All students in a university
- 🏠 All houses in a city
- 🛒 All customers of a company
- 🚗 All cars produced by a manufacturer
- 📱 All users of an application
- 🏥 All patients treated by a hospital
- 📊 All observations in a dataset

### 💡 Simple Example

Suppose a college has **10,000 students** and we want to know the average height of all students.

Then:

> **Population = All 10,000 students**

The population represents the complete group we are interested in.

## 🎯 What is a Sample?

A **Sample** is a smaller group selected from the population for analysis.

Instead of studying every member of the population, we select a smaller number of observations.

### 💡 Example

Suppose our college has:

**Population = 10,000 students**

We randomly select:

**Sample = 500 students**

We can study the heights of those 500 students and use the results to learn something about the larger population.

### 🔑 Key Idea

> **Population = Entire group**

> **Sample = Smaller group selected from the population**

## ⚖️ Population vs Sample

| 👥 Population | 🎯 Sample |
|---|---|
| Entire group | Part of the population |
| Usually larger | Usually smaller |
| Contains all observations of interest | Contains selected observations |
| Can be expensive to study | Usually cheaper to study |
| Can take more time | Usually faster to analyze |
| Used to calculate parameters | Used to calculate statistics |

### Example

If a company has **1,000,000 customers**:

- 👥 Population → All 1,000,000 customers
- 🎯 Sample → 5,000 selected customers

The sample should ideally represent the important characteristics of the population.

## 🔍 Why Do We Use Samples?

Studying an entire population is not always practical.

There are several reasons why sampling is useful.

### ⏱️ 1. Saves Time

Analyzing a smaller number of observations is generally faster.

### 💰 2. Reduces Cost

Collecting information from every member of a population can be expensive.

### 📊 3. Easier Analysis

Smaller datasets are easier to manage and analyze.

### 🌍 4. Population May Be Too Large

Some populations are extremely large.

For example:

- All internet users
- All transactions of a global company
- All images generated by a system

Studying every observation may not be practical.

### 🧪 5. Some Measurements Are Destructive

In some experiments, testing an object may destroy it.

For example, a manufacturer may test the strength of products.

Testing every product would not be practical.

### 🔑 Key Idea

> **Sampling allows us to learn about a large population using a smaller amount of data.**

## 📐 What is a Parameter?

A **Parameter** is a numerical value that describes a characteristic of the entire population.

Examples include:

- Population mean
- Population variance
- Population standard deviation
- Population proportion

### Example

Suppose the average height of all 10,000 students in a college is:

**170 cm**

If all 10,000 students are considered, then:

> 170 cm is a **Population Parameter**.

### Common Symbols

Population mean:

**μ (mu)**

Population standard deviation:

**σ (sigma)**

Population variance:

**σ²**

## 📊 What is a Statistic?

A **Statistic** is a numerical value calculated from a sample.

Examples include:

- Sample mean
- Sample variance
- Sample standard deviation
- Sample proportion

### Example

Suppose we select 500 students from a college and calculate their average height.

The result is:

**168.5 cm**

Because this value was calculated from the sample:

> 168.5 cm is a **Sample Statistic**.

### Common Symbols

Sample mean:

**x̄ (x-bar)**

Sample standard deviation:

**s**

Sample variance:

**s²**

## 🧩 Parameter vs Statistic

| 📐 Parameter | 📊 Statistic |
|---|---|
| Describes a population | Describes a sample |
| Calculated from population data | Calculated from sample data |
| Usually represented by Greek symbols | Often represented by Latin symbols |
| Population mean → μ | Sample mean → x̄ |
| Population standard deviation → σ | Sample standard deviation → s |
| Population variance → σ² | Sample variance → s² |

### 💡 Easy Way to Remember

**Population → Parameter**

**Sample → Statistic**

Both describe numerical characteristics, but they come from different groups.

## 🔄 From Population to Sample

The basic idea of statistical sampling can be represented as:

**👥 Population**

↓  
Select observations

↓  

**🎯 Sample**

↓  
Calculate a statistic

↓  

**📊 Sample Information**

↓  
Use it to learn about the population

This process is the foundation of **Inferential Statistics**.

The goal is not simply to collect a small dataset.

The goal is to select a sample that provides useful information about the larger population.

## 🧪 Simple Example

Suppose we have the following population of exam scores:

`[50, 55, 60, 65, 70, 75, 80, 85, 90, 95]`

Here:

**Population = 10 students**

Now suppose we select:

`[55, 65, 75, 85, 95]`

This becomes our sample.

Therefore:

- 👥 Population size = 10
- 🎯 Sample size = 5

We can calculate the population mean and sample mean and compare them.

In [1]:
import numpy as np

population = np.array([50, 55, 60, 65, 70, 75, 80, 85, 90, 95])

sample = np.array([55, 65, 75, 85, 95])

population_mean = np.mean(population)
sample_mean = np.mean(sample)

print("Population Mean:", population_mean)
print("Sample Mean:", sample_mean)

Population Mean: 72.5
Sample Mean: 75.0


## 🔍 Understanding the Difference

The population mean is:

**μ = 72.5**

The sample mean is:

**x̄ = 75.0**

They are not exactly the same.

This is completely normal.

A sample is only a subset of the population, so its statistics may differ from the true population parameters.

This difference is one reason why **sampling uncertainty** exists.

### 🔑 Important Idea

> A sample statistic is used to estimate a population parameter.

As we learn more about sampling and statistical inference, we will understand how to quantify this uncertainty.

## 🎲 Random Sampling

One important idea in sampling is **random selection**.

Random sampling gives members of the population a chance of being selected.

For example, instead of manually choosing students we think are representative, we can randomly select students from the population.

This helps reduce the influence of personal selection bias.

### Example

Suppose we have 1,000 customers.

Instead of selecting only our most active customers, we randomly select 100 customers.

This gives us a better chance of obtaining a sample that represents different types of customers.

In [2]:
np.random.seed(42)

population = np.arange(1, 101)

sample = np.random.choice(
    population,
    size=10,
    replace=False
)

print("Population:", population)
print("Sample:", sample)

Population: [  1   2   3   4   5   6   7   8   9  10  11  12  13  14  15  16  17  18
  19  20  21  22  23  24  25  26  27  28  29  30  31  32  33  34  35  36
  37  38  39  40  41  42  43  44  45  46  47  48  49  50  51  52  53  54
  55  56  57  58  59  60  61  62  63  64  65  66  67  68  69  70  71  72
  73  74  75  76  77  78  79  80  81  82  83  84  85  86  87  88  89  90
  91  92  93  94  95  96  97  98  99 100]
Sample: [84 54 71 46 45 40 23 81 11  1]


### 🔍 What Happened?

We created a population containing values from **1 to 100**.

Then we randomly selected **10 observations** without replacement.

So:

- 👥 Population size = 100
- 🎯 Sample size = 10
- 🔄 Sampling method = Random sampling
- 🚫 Replacement = No

Random sampling is one of the fundamental ideas we will explore further in the **Sampling and CLT** section of this repository.

## 📊 Sample Size

**Sample size** refers to the number of observations included in a sample.

For example:

- Sample of 10 → `n = 10`
- Sample of 100 → `n = 100`
- Sample of 1,000 → `n = 1000`

A larger sample can often provide more information about a population.

However:

> **Bigger is not automatically better.**

The quality and representativeness of the sample are also important.

A very large but biased sample can still produce misleading conclusions.

### 🔑 Important

Good statistical analysis depends on both:

**Sample Size + Sample Quality**

## 🚨 Sampling Bias

**Sampling Bias** occurs when the selected sample does not properly represent the population.

### Example

Suppose a company wants to know how satisfied all customers are.

There are:

- 90,000 normal customers
- 10,000 premium customers

If the company surveys only premium customers, the results may not represent the entire customer population.

The problem is not necessarily the sample size.

The problem is **how the sample was selected**.

### 💡 Key Idea

> A sample should represent the population as well as possible.

Poor sampling can lead to misleading conclusions, even when the statistical calculations are correct.

## 🤖 Population and Sample in AI/ML

Population and Sample concepts are highly relevant to Machine Learning.

### 📊 Dataset Creation

A real-world population may be extremely large.

For example:

> All possible customers who could use an online shopping platform.

We may only have access to a subset of those customers.

That observed dataset acts as a sample from a broader population.

### 🧪 Model Training

Machine Learning models learn from available data.

We commonly divide available data into:

- Training data
- Validation data
- Test data

These subsets help us evaluate how well a model generalizes beyond the observations it was trained on.

### 📈 Model Evaluation

If a model performs well on training data but poorly on unseen data, it may not generalize well.

This connects directly to the statistical idea of learning from observed data and making conclusions about unseen observations.

### 🎯 Real-World Goal

> **We ultimately care about how well our model performs on data beyond the specific observations used during training.**

## 🔗 Statistics → Machine Learning Connection

| 📊 Statistical Concept | 🤖 Machine Learning Connection |
|---|---|
| Population | Broader real-world data |
| Sample | Available dataset |
| Parameter | True population characteristic |
| Statistic | Value calculated from observed data |
| Sampling | Dataset selection |
| Sampling Bias | Dataset bias |
| Sample Size | Amount of available information |
| Random Sampling | More representative data collection |
| Inferential Statistics | Generalizing from observed data |
| Uncertainty | Understanding model reliability |

### 💡 Big Picture

In Machine Learning, we usually cannot collect every possible observation.

Instead, we learn from available data and hope that our data provides useful information about the broader problem.

This is why understanding **Population, Sample, and Sampling** is important for Machine Learning.

## 💎 Key Insights

- 👥 **Population** = Entire group we are interested in studying.
- 🎯 **Sample** = Smaller group selected from the population.
- 📐 **Parameter** = Numerical characteristic of a population.
- 📊 **Statistic** = Numerical characteristic calculated from a sample.
- ⏱️ Sampling can save time and resources.
- 🎲 Random sampling can help reduce selection bias.
- 📊 Sample size matters, but sample quality matters too.
- 🚨 A biased sample can produce misleading conclusions.
- 🤖 Machine Learning often learns from a limited set of observations and must generalize to unseen data.

### ⭐ Remember

> **Population → Sample → Statistic → Inference about Population**

## 📝 Practice Questions

### 🧠 Conceptual Questions

1. What is a Population?
2. What is a Sample?
3. What is the difference between a Parameter and a Statistic?
4. Why do we use samples instead of studying entire populations?
5. What is sampling bias?
6. Why is random sampling useful?
7. Is a larger sample always better? Explain.

---

### 💻 Practical Questions

1. Create a population containing the ages of 100 people.
2. Randomly select 20 people from the population.
3. Calculate the population mean.
4. Calculate the sample mean.
5. Compare the two values.
6. Repeat the sampling process with a different sample.
7. Observe how the sample mean changes.

---

### 🤖 AI/ML Thinking

Suppose an ML model is trained using data collected only from customers in one city.

Answer:

1. What could be considered the population?
2. What could be considered the sample?
3. What kind of sampling problem might occur?
4. How could this affect model performance on customers from other cities?
5. Why is representative data important for Machine Learning?

## 📌 Summary

In this notebook, we learned the foundation of sampling in Statistics.

### We covered:

- 👥 Population
- 🎯 Sample
- 📐 Parameter
- 📊 Statistic
- ⚖️ Population vs Sample
- 🔍 Why sampling is necessary
- 🎲 Random sampling
- 📊 Sample size
- 🚨 Sampling bias
- 🤖 Population and Sample in Machine Learning

### 🚀 Final Takeaway

> **We often cannot study an entire population, so we use samples to learn about it.**

The quality of our conclusions depends heavily on how well the sample represents the population.

Understanding this idea is essential before moving into:

**Sampling Techniques → Sampling Distributions → Central Limit Theorem → Statistical Inference**

---

### 🔜 What's Next?

In the next notebook, we will learn about:

**Types of Data**

and understand how data can be classified into different categories and why the type of data matters when choosing statistical methods.